include/clang/AST/Expr.h

```diff
@@ -2962,6 +2962,77 @@ class CStyleCastExpr final
   friend class CastExpr;
 };
 
+class BoundsCastExpr final
+    : public ExplicitCastExpr,
+      private llvm::TrailingObjects<BoundsCastExpr, CXXBaseSpecifier *> {
+  SourceLocation LPLoc;
+  SourceLocation RPLoc;
+  Expr *BaseExpr = nullptr;
+  Expr *CountExpr = nullptr;
```

Comment:
I think it is a good idea to have a separate class and to not combine this with CStyleCast.